kytos/core/auth.py

```diff
@@ -49,13 +53,17 @@ class Auth:
 
     encode_algorithm = "HS256"
 
-    def __init__(self, controller):
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, controller, get_mongo=lambda: Mongo()):
         """Init method of Auth class takes the parameters below.
 
         Args:
             controller(kytos.core.controller): A Controller instance.
 
         """
+        self.mongo = get_mongo()
```

Comment:
@Alopalao, to be aligned with how DB controllers are structured in the codebase, please let's extract these new methods that you've added here on `UserController`, and then `Auth` uses and depends on `UserController`, segregating DB and API endpoints responsibilities.